EasyPost.Tests/BatchTest.cs

```diff
@@ -40,154 +43,159 @@ private static async Task<Batch> CreateOneCallBuyBatch()
-                {
-                    "shipments", new List<Dictionary<string, object>>
-                    {
-                        Fixture.OneCallBuyShipment
-                    }
-                }
+                shipment
             });
 
-            Assert.IsInstanceOfType(batch, typeof(Batch));
-            Assert.IsTrue(batch.id.StartsWith("batch_"));
-            Assert.AreEqual(1, batch.num_shipments);
+            Assert.Equal(1, batch.num_shipments);
+
+            batch = await batch.RemoveShipments(new List<Shipment>
+            {
+                shipment
+            });
+
+            Assert.Equal(0, batch.num_shipments);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestBuy()
         {
-            _vcr.SetUpTest("buy");
+            UseVCR("buy");
 
             Batch batch = await CreateOneCallBuyBatch();
 
-            await batch.Buy();
+            batch = await batch.Buy();
 
-            Assert.IsInstanceOfType(batch, typeof(Batch));
-            Assert.AreEqual(1, batch.num_shipments);
+            Assert.IsType<Batch>(batch);
+            Assert.Equal(1, batch.num_shipments);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestCreateScanForm()
         {
-            _vcr.SetUpTest("create_scan_form");
+            UseVCR("create_scan_form");
 
 
             Batch batch = await CreateOneCallBuyBatch();
 
-            await batch.Buy();
-
-            if (_vcr.IsRecording())
+            if (IsRecording())
             {
                 Thread.Sleep(10000); // Wait enough time to process
             }
 
-            await batch.GenerateScanForm();
+            batch = await batch.Buy();
+
+            if (IsRecording())
+            {
+                Thread.Sleep(30000); // Wait enough time to process
+            }
+
+            batch = await batch.GenerateScanForm();
 
             // We can't assert anything meaningful here because the scanform gets queued for generation and may not be immediately available
-            Assert.IsInstanceOfType(batch, typeof(Batch));
+            Assert.IsType<Batch>(batch);
         }
 
-        [TestMethod]
-        public async Task TestAddRemoveShipment()
+        [Fact]
+        [CrudOperations.Update]
+        public async Task TestLabel()
         {
-            _vcr.SetUpTest("add_remove_shipment");
+            UseVCR("label");
 
-            Shipment shipment = await Shipment.Create(Fixture.OneCallBuyShipment);
+            Batch batch = await CreateOneCallBuyBatch();
 
-            Batch batch = await Batch.Create();
+            batch = await batch.Buy();
 
-            await batch.AddShipments(new List<Shipment>
+            if (IsRecording())
             {
-                shipment
-            });
-
-            Assert.AreEqual(1, batch.num_shipments);
+                Thread.Sleep(30000); // Wait enough time to process
```

Comment:
Same comment as above ^

Reply:
EDIT: After looking, it seems you reduced these to 10s. Please also mention what you did when responding to feedback. It greatly reduces the back and forth instead of requiring a reviewer to dig into each change individually.